### PR TITLE
Fix build: migrate from ILRepack.Lib.MSBuild.Task to ILRepack.Lib.MSBuild

### PR DESCRIPTION
**Pull request by lbmaian**
#335 included migrating from `ILRepack.MSBuild.Task` to `ILRepack.Lib.MSBuild.Task`, since the former is no longer maintained (its GitHub repo has been archived) and the latter drastically reduces logging that included warnings on non-Windows.

However, although the project built locally in Visual Studio, it broke `dotnet` builds due to `ILRepack.Lib.MSBuild.Task`'s usage of outdated `Microsoft.Build.Utilities.v4.0`.

I found a more maintained ILRepack build task that does work with `dotnet`, yet keeps the benefit of reduced logging:  `ILRepack.Lib.MSBuild`. So this migrates from `ILRepack.Lib.MSBuild.Task` to `ILRepack.Lib.MSBuild`.

As a bonus, `ILRepack.Lib.MSBuild` is both easier to use and uses a more maintained fork of `ILRepack`/`ILRepack.Lib`.

### CODE REVIEW

**Comment by lbmaian** [comment]
I had to workaround a problem with macOS builds requiring a DYLD_* env var to be set (since it's unset likely due to https://developer.apple.com/library/archive/documentation/Security/Conceptual/System_Integrity_Protection_Guide/RuntimeProtections/RuntimeProtections.html).

File: Harmony/Harmony.csproj [hunk 1]
```diff
@@ -64,17 +97,46 @@
 	<ItemGroup>
 		<!-- https://dev.azure.com/MonoMod/MonoMod/_packaging?_a=feed&feed=DevBuilds -->
 		<PackageReference Include="MonoMod.Common" Version="20.5.31.1">
-			<PrivateAssets Condition="!$(IsCoreOrStandard) And '$(Configuration)'=='Release'">All</PrivateAssets>
+			<PrivateAssets Condition="$(ILRepackInternalize)">all</PrivateAssets>
```
Comment (pardeike) [inline Harmony/Harmony.csproj:100]: Nice with an extra variable here!

**Review by pardeike — APPROVED** [review]

**Comment by pardeike** [comment]
Since it works, can we merge to master and fix subsequent issues in a new PR?

**Comment by lbmaian** [comment]
Sure. Only change I've made since is just updating some comments.

**Comment by lbmaian** [comment]
There will be another followup PR since it turns out `ILRepack.Lib.MSBuild` is also problematic. It uses a fork of ILRepack that changes the internalization algorithm that results in some of non-Harmony types no longer being internalized.

**Pull request merged**